libsolutil/IpfsHash.cpp

```diff
@@ -53,36 +53,133 @@ string base58Encode(bytes const& _data)
 	reverse(output.begin(), output.end());
 	return output;
 }
+struct Chunk
+{
+	Chunk() = default;
+	Chunk(bytes _hash, size_t _size, size_t _blockSize):
+		hash(std::move(_hash)),
+		size(_size),
+		blockSize(_blockSize),
+		links{},
+		lengths{}
+	{}
+	bytes hash = {};
+	size_t size = 0;
+	size_t blockSize = 0;
+
+	bytes links = {};
+	size_t linksBlockSize = 0;
+	bytes lengths = {};
+
+	void addLink(Chunk& _chunkLink)
+	{
+		size += _chunkLink.size;
+		linksBlockSize += _chunkLink.blockSize;
+
+		bytes newLinkData =
+			bytes {0x0a} +
+			varintEncoding(_chunkLink.hash.size()) +
+			std::move(_chunkLink.hash) +
+			bytes{0x12, 0x00, 0x18} + varintEncoding(_chunkLink.blockSize);
+
+		links += bytes{0x12} + varintEncoding(newLinkData.size()) + std::move(newLinkData);
+		lengths += bytes{0x20} + varintEncoding(_chunkLink.size);
+	}
+
+	void updateHash()
+	{
+		bytes tmp = bytes{0x08, 0x02, 0x18} + varintEncoding(size) + lengths;
+		bytes blockData = links + bytes{0x0a} + varintEncoding(tmp.size()) + tmp;
```

Comment:
Can you create a helper function `encodeByteArray(bytes const& _data) { return bytes{0x0a} + varintEncoding(_data.size()) + _data; }`?

Reply:
Created, added also one more reusable function to reduce code repetition.